swarm/api/api.go

```diff
@@ -654,72 +657,79 @@ func (self *Api) BuildDirectoryTree(mhash string, nameresolver bool) (addr stora
-		_, err = self.resource.LookupHistorical(ctx, rsrc.NameHash(), period, true, maxLookup)
+		_, err = a.resource.LookupHistorical(ctx, rsrc.NameHash(), period, true, maxLookup)
 	} else {
-		_, err = self.resource.LookupLatest(ctx, rsrc.NameHash(), true, maxLookup)
+		_, err = a.resource.LookupLatest(ctx, rsrc.NameHash(), true, maxLookup)
 	}
 	if err != nil {
 		return "", nil, err
 	}
 	var data []byte
-	_, data, err = self.resource.GetContent(rsrc.NameHash().Hex())
+	_, data, err = a.resource.GetContent(rsrc.NameHash().Hex())
 	if err != nil {
 		return "", nil, err
 	}
 	return rsrc.Name(), data, nil
 }
 
-func (self *Api) ResourceCreate(ctx context.Context, name string, frequency uint64) (storage.Address, error) {
-	key, _, err := self.resource.New(ctx, name, frequency)
+// ResourceCreate - create's Resource and returns it's key
+func (a *API) ResourceCreate(ctx context.Context, name string, frequency uint64) (storage.Address, error) {
+	key, _, err := a.resource.New(ctx, name, frequency)
 	if err != nil {
 		return nil, err
 	}
 	return key, nil
 }
 
-func (self *Api) ResourceUpdateMultihash(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
-	return self.resourceUpdate(ctx, name, data, true)
+// ResourceUpdateMultihash - updates Multihash resource
+func (a *API) ResourceUpdateMultihash(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
+	return a.resourceUpdate(ctx, name, data, true)
 }
-func (self *Api) ResourceUpdate(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
-	return self.resourceUpdate(ctx, name, data, false)
+
+// ResourceUpdate - for non 'Multihash' resource
+func (a *API) ResourceUpdate(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
+	return a.resourceUpdate(ctx, name, data, false)
 }
 
-func (self *Api) resourceUpdate(ctx context.Context, name string, data []byte, multihash bool) (storage.Address, uint32, uint32, error) {
+func (a *API) resourceUpdate(ctx context.Context, name string, data []byte, multihash bool) (storage.Address, uint32, uint32, error) {
 	var addr storage.Address
 	var err error
 	if multihash {
-		addr, err = self.resource.UpdateMultihash(ctx, name, data)
+		addr, err = a.resource.UpdateMultihash(ctx, name, data)
 	} else {
-		addr, err = self.resource.Update(ctx, name, data)
+		addr, err = a.resource.Update(ctx, name, data)
 	}
-	period, _ := self.resource.GetLastPeriod(name)
-	version, _ := self.resource.GetVersion(name)
+	period, _ := a.resource.GetLastPeriod(name)
+	version, _ := a.resource.GetVersion(name)
 	return addr, period, version, err
 }
 
-func (self *Api) ResourceHashSize() int {
-	return self.resource.HashSize
+// ResourceHashSize - accessor
+func (a *API) ResourceHashSize() int {
+	return a.resource.HashSize
 }
 
-func (self *Api) ResourceIsValidated() bool {
-	return self.resource.IsValidated()
+// ResourceIsValidated - accessor
```

Comment:
ResourceIsValidated check if the resource is validated

Reply:
ResourceIsValidated checks if the Mutable Resource has an active content validator.